SimPEG/VRM/RxVRM.py

```diff
@@ -20,19 +20,65 @@ def __init__(self, locs, **kwargs):
 class Point(BaseRxVRM):
     """Point receiver"""
 
-    def __init__(self, locs, times, fieldType, fieldComp, **kwargs):
+    times = properties.Array('Observation times', dtype=float)
+    fieldType = properties.StringChoice('Field type', choices=["h", "b", "dhdt", "dbdt"])
+    fieldComp = properties.StringChoice('Component of response', choices=["x", "y", "z"])
+
+    # def __init__(self, locs, times, fieldType, fieldComp, **kwargs):
+    def __init__(self, locs, **kwargs):
         assert locs.shape[1] == 3, 'locs must in 3-D (x,y,z).'
         super(Point, self).__init__(locs, **kwargs)
-        self.times = times
-        assert fieldType in ['h', 'b', 'dhdt', 'dbdt'], '"fieldType" must be one of "h", "b", "dhdt" or "dbdt"'
-        self.fieldType = fieldType
-        assert fieldComp in ['x', 'y', 'z'], '"fieldComp" must be one of "x", "y" or "z"'
-        self.fieldComp = fieldComp
 
     @property
     def nTimes(self):
         """Number of measurements times."""
-        return len(self.times)
+        if self.times is not None:
+            return len(self.times)
+
+    @property
+    def nLocs(self):
+        """Number of locations."""
+        return self.locs.shape[0]
+
+    @property
+    def nD(self):
+        """Number of data in the receiver."""
+        if self.times is not None:
+            return self.locs.shape[0] * len(self.times)
+
+
+class SquareLoop(BaseRxVRM):
+    """Square loop receiver
+
+    Measurements with this type of receiver are the field, integrated over the
+    area of the loop, then multiplied by the number of coils, then normalized
+    by the dipole moment. As a result, the units for fields predicted with this
+    type of receiver are the same as 'h', 'b', 'dhdt' and 'dbdt', respectively.
+
+    """
+
+    times = properties.Array('Observation times', dtype=float)
+    width = properties.Float('Square loop width', min=1e-6)
+    nTurns = properties.Integer('Number of loop turns', min=1, default=1)
+    quadOrder = properties.Integer(
+        'Order for numerical quadrature integration over loop', min=1, max=7, default=3
+        )
+    fieldType = properties.StringChoice('Field type', choices=["h", "b", "dhdt", "dbdt"])
+    fieldComp = properties.StringChoice('Component of response', choices=["x", "y", "z"])
+
+    # def __init__(self, locs, times, width, nTurns, fieldType, fieldComp, **kwargs):
+    def __init__(self, locs, **kwargs):
+
+        # self._quad_order = kwargs.get('quad_order', 4)
+
+        assert locs.shape[1] == 3, 'locs must in 3-D (x,y,z).'
+        super(SquareLoop, self).__init__(locs, **kwargs)
```

Comment:
The second observation goes along with a chat I had with @lindsey on Slack just yesterday. There are a lot of `assert` in SimPEG, for historical reasons. However, for new code I would suggest to change them all to `if -raise`, as `assert` is not really for input checking, but for debugging and testing.

So instead of `assert locs.shape...` do:
```
if locs.shape[1] != 3:
    raise ValueError('locs must in 3-D (x,y,z).')
````
or even better (more informative):
```
if locs.shape[1] != 3:
    raise ValueError('locs must in 3-D (x,y,z), but only %i dimensions provided.' % locs.shape[1])
````
Again, this might be just me, I am not 100 % sure of the opinion of @lheagy, @rowanc1, @sgkang et. al.

Reply:
Thanks @prisae: completely agree - this is absolutely the way we should be going!